### PR TITLE
swarm: client-side MRU signatures ; BMT fixes ; network simulation tests

### PR DESCRIPTION
**Pull request by nonsense**
This PR is merging upstream changes on `swarm` over the last few weeks, namely:
- client-side MRU signatures (@jpeletier)
- BMT fixes (@zelig)
- network simulation tests (@janos , @holisticode)

Once this is merged, we will try a development flow, where we merge every PR in `ethereum/go-ethereum`, rather than keep a more up-to-date `ethersphere` fork.

### CODE REVIEW

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 872defcc83ad4a3c785d02dcf70df59e898370cd
  * Your commit message body contains a line that is longer than 80 characters
  * Commits must be prefixed with the package(s) they modify

 * Commit: 1929da347aa7d009baa1a4392032a555907018ea
  * Your commit message body contains a line that is longer than 80 characters
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 2743b69ded03948f2ad3ba91b4fc30b8e5243f1c
  * Commits must be prefixed with the package(s) they modify

 * Commit: 2ec65a57d7918ea991a5a49b78005cacc3232329
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

File: swarm/testutil/http.go [hunk 1]
```diff
@@ -97,12 +99,25 @@ func NewTestSwarmServer(t *testing.T, serverFunc func(*api.API) TestServer) *Tes
...
 
 func (t *TestSwarmServer) Close() {
 	t.cleanup()
 }
+
+func (t *TestSwarmServer) GetCurrentTime() mru.Timestamp {
+	return t.timestampProvider.Now()
+}
+
+// EnableMetrics is starting InfluxDB reporter so that we collect stats when running tests locally
```
Comment (nonsense) [inline swarm/testutil/http.go]: @jpeletier isn't this creating a circular dependency? Are you actually using `EnableMetrics` anywhere?

Comment (jpeletier) [reply swarm/testutil/http.go]: ?? I didn't touch anything regarding that EnableMetrics. Only the code above `GetCurrentTime()` belongs to this change. Maybe @nolash knows.

Comment (nonsense) [reply swarm/testutil/http.go]: OK, I will remove it.

**Comment by nonsense** [comment]
@zelig this is touching only `swarm` and `p2p/simulations`. Tests are passing, and this has been approved already on `ethersphere`. Please merge when you get a chance.

**Pull request merged**